NODE_OPTIONS handling breaks with quoted arguments

<!---
Thanks for filing an issue 😄 ! Before you submit, please read the following:

Search open/closed issues before submitting since someone might have asked the same thing before!
-->

# 🐛 bug report

Running parcel under the VS Code debugger or under VS Code with Auto Attach enabled causes it to break with the error:

```
@parcel/workers: Cannot find module '"c:/Users/321/AppData/Local/Programs/Microsoft'
Require stack:
- internal/preload
Error: Cannot find module '"c:/Users/1/AppData/Local/Programs/Microsoft'
Require stack:
- internal/preload
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:892:15)
    at Function.Module._load (internal/modules/cjs/loader.js:742:27)
    at Module.require (internal/modules/cjs/loader.js:964:19)
    at Module._preloadModules (internal/modules/cjs/loader.js:1221:12)
    at loadPreloadModules (internal/bootstrap/pre_execution.js:446:5)
    at MessagePort.<anonymous> (internal/main/worker_thread.js:122:5)
    at MessagePort.[nodejs.internal.kHybridDispatch] (internal/event_target.js:353:41)   
    at MessagePort.exports.emitMessage (internal/per_context/messageport.js:18:26) 
```

## 🎛 Configuration (.babelrc, package.json, cli command)

N/A

## 🤔 Expected Behavior

It doesn't break!

## 😯 Current Behavior

It breaks!

## 💁 Possible Solution

It looks like the handling here is improper

https://github.com/parcel-bundler/parcel/blob/97a7bfeead8007df257d13d94c5b2f724a2ba0bd/packages/core/workers/src/Worker.js#L64-L74

This doesn't deal with quotes arguments, which then leads to a malformed environment. This can be seen easily by extracting that snippet to a test script:

![](https://memes.peet.io/img/21-04-7f179e81-b378-41c6-95e7-c91aee91856d.png)

## 🔦 Context

This was originally reported in https://github.com/microsoft/vscode-js-debug/issues/967

## 💻 Code Sample

N/A. Here's docs for auto attach if you want to try turning it on to reproduce locally: https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_auto-attach

--- a/packages/core/workers/src/Worker.js
+++ b/packages/core/workers/src/Worker.js
@@ -4,6 +4,7 @@ import type {FilePath} from '@parcel/types';
 import type {BackendType, WorkerImpl, WorkerMessage} from './types';
 import type {SharedReference} from './WorkerFarm';
 
+import nullthrows from 'nullthrows';
 import EventEmitter from 'events';
 import ThrowableDiagnostic from '@parcel/diagnostic';
 import {getWorkerBackend} from './backend';
@@ -63,7 +64,19 @@ export default class Worker extends EventEmitter {
 
     // Workaround for https://github.com/nodejs/node/issues/29117
     if (process.env.NODE_OPTIONS) {
-      let opts = process.env.NODE_OPTIONS.split(' ');
+      // arg parsing logic adapted from https://stackoverflow.com/a/46946420/2352201
+      let opts = [''];
+      let quote = false;
+      for (let c of nullthrows(process.env.NODE_OPTIONS.match(/\\?.|^$/g))) {
+        if (c === '"') {
+          quote = !quote;
+        } else if (!quote && c === ' ') {
+          opts.push('');
+        } else {
+          opts[opts.length - 1] += c.replace(/\\(.)/, '$1');
+        }
+      }
+
       for (let i = 0; i < opts.length; i++) {
         let opt = opts[i];
         if (opt === '-r' || opt === '--require') {

